infrastructure/sentry-subgraph/src/utils/updateTotalAccruedRewards.ts

```diff
@@ -0,0 +1,30 @@
+import { SentryKey, PoolInfo, SentryWallet } from "../../generated/schema";
+import { Address, BigInt, ethereum, log } from "@graphprotocol/graph-ts"
+
+/**
+ * 
+ * Update the total Accrued esXai for either the sentryWallet or the staking pool for key based submissions
+ * 
+*/
+export function updateTotalAccruedRewards(sentryKey: SentryKey, reward: BigInt, event: ethereum.Event): void {
+  if (sentryKey.assignedPool.toHexString() == new Address(0).toHexString()) {
+    // Rewards claimed to the owner of the key
+    const sentryWallet = SentryWallet.load(sentryKey.sentryWallet);
+    if (!sentryWallet) {
+      log.warning("Failed to find sentryWallet handleAssertionSubmitted: keyID: " + sentryKey.keyId.toString() + ", TX: " + event.transaction.hash.toHexString(), []);
+      return;
+    }
+    sentryWallet.totalAccruedAssertionRewards = sentryWallet.totalAccruedAssertionRewards.plus(reward)
+    sentryWallet.save()
+
+  } else {
```

Comment:
Same as above, I think we could remove this else and just return here?

Otherwise looks good. I'm going to approve as is. Feel free to change if you like, or merge!

Reply:
Not sure I understand, the else handles the case for the rewards going to the pool ?